pymatgen/analysis/graphs.py

```diff
@@ -1597,7 +1597,7 @@ def with_empty_graph(cls, molecule, name="bonds", edge_weight_name=None, edge_we
         return cls(molecule, graph_data=graph_data)
 
     @staticmethod
-    def with_edges(molecule: Molecule, edges: dict[tuple[int, int], dict]):
+    def with_edges(molecule: Molecule, edges: dict[tuple[int, int], None | dict]) -> MoleculeGraph:
```

Comment:
type annotation of `edges` was likely wrong. Apparently the values could be `None`:
```python
for edge, props in edges.items():
            try:
                from_index = edge[0]
                to_index = edge[1]
            except TypeError:
                raise ValueError("Edges must be given as (from_index, to_index) tuples")

            if props is None:
                weight = None

            else:
                weight = props.pop("weight", None)
                if len(props.items()) == 0:
                    props = None
```

But still get `pymatgen/analysis/fragmenter.py:86: error: Argument 2 to "with_edges" of "MoleculeGraph" has incompatible type "dict[tuple[Any, Any], None]"; expected "dict[tuple[int, int], Optional[dict[Any, Any]]]"  [arg-type]`